datafusion/expr/src/simplify.rs

```diff
@@ -41,40 +41,12 @@ pub trait SimplifyInfo {
 
     /// Returns data type of this expr needed for determining optimized int type of a value
     fn get_data_type(&self, expr: &Expr) -> Result<DataType>;
+
+    /// Return the schema for function simplifier
```

Comment:
If we are going to move the SimplifyInfo, I think we should now also remove the schema requirement from this API

Reply:
Any reason why we should avoid return schema? Like `cast_to` is the function that require schema, it's possible that there are other functions with similar requirements.